File: app/Models/ContentAgentRequest.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class ContentAgentRequest extends Model
{
    protected $table = 'content_agent_requests';

    protected $fillable = [
        'request_id',
        'workflow_type',
        'backend',
        'engine_hint',
        'status',
        'correlation_type',
        'correlation_id',
        'contract_version',
        'payload_json',
        'result_json',
        'error_message',
        'submitted_at',
        'completed_at',
    ];

    protected function casts(): array
    {
        return [
            'correlation_id' => 'integer',
            'submitted_at' => 'datetime',
            'completed_at' => 'datetime',
        ];
    }
}

File: geoflow-v3/backend/app/models/content_agent_request.py
from datetime import datetime

from sqlalchemy import BigInteger, DateTime, JSON, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class ContentAgentRequest(Base):
    __tablename__ = "content_agent_requests"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True, autoincrement=True)
    request_id: Mapped[str] = mapped_column(String(64), nullable=False, unique=True)
    workflow_type: Mapped[str] = mapped_column(String(50), nullable=False)
    status: Mapped[str] = mapped_column(String(30), default="pending")
    engine: Mapped[str] = mapped_column(String(30), default="")
    correlation_type: Mapped[str] = mapped_column(String(50), default="")
    correlation_id: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    error_message: Mapped[str] = mapped_column(Text, default="")
    result_json: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    submitted_at: Mapped[datetime | None] = mapped_column(DateTime, server_default=func.now())
    completed_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    created_at: Mapped[datetime | None] = mapped_column(DateTime, server_default=func.now())
    updated_at: Mapped[datetime | None] = mapped_column(DateTime, server_default=func.now(), onupdate=func.now())
